corehq/messaging/smsbackends/telerivet/tasks.py

```diff
@@ -44,3 +46,32 @@ def process_incoming_message(*args, **kwargs):
                 domain_scope=domain_scope, backend_id=backend.couch_id)
         elif kwargs["message_type"] == MESSAGE_TYPE_CALL:
             log_call(from_number, "TELERIVET-%s" % kwargs["message_id"], backend=backend)
+
+
+@task(serializer='pickle', queue=CELERY_QUEUE, ignore_result=True)
+def process_message_status(message_id, status, request_secret, **kwargs):
```

Comment:
Can we avoid having `request_secret` being passed as the argument? I am guessing this a password-kind and having that as arguments being passed around/stored in RabbitMQ and then celery would not be nice.

Reply:
Generally here we are okay with getting a backend that does not exist or if the sms object is not found or if the status is not one of the many that we expect, instead of raising an error, is that intentional?

I feel the backend check should simply just go in the view and give back a 403 if the secret isn't valid (unless that is a security concern where one could brute force, highly unlikely I believe) and pass the backend ID to this method (to also solve the above point).
Same should be for "status".
If there is no SMS, this should also give an error, preferably on the view itself.

I understand we want the view to be as minimal as possible but there is currently no mechanism for telerivent to know that their request to update status was even valid/accepted or if there is something they/we need to fix for example if they added a new status message that we have not added yet or if they made a change where they missed passing the secret properly. We are silently absorbing failures which would go unnoticed for quite a while.

I am not sure if the code suggestions above are the best way, but my concern is simply better comms between APIs.

Reply:
@mkangia Any other status than 200 will be treated as failure according to their docs. They don't seem to specify what their course of action is when they made a failed attempt (retry or whatever).